### DISCUSSION TITLE
does anyone know how to do crc32 on ipv6hdr->saddr?

### DISCUSSION THREAD

**Discussion by kolinfluence** [cilium/ebpf#884] [Q&A] [answered]
does anyone know how to do crc32 on ipv6hdr->saddr?
i cant convert ipv6hdr->saddr to something crc32 can digest. how shld i do this?
> tcphdr->dest = bpf_htons(80 + (ipv6hdr->saddr % nr_cpus)); 

```
	if (eth_type == bpf_htons(ETH_P_IP)) {
		ip_type = parse_iphdr(&nh, data_end, &iphdr);

				tcphdr->dest = bpf_htons(80 + (iphdr->saddr % nr_cpus));

	} else if (eth_type == bpf_htons(ETH_P_IPV6)) {
		
		ip_type = parse_ip6hdr(&nh, data_end, &ipv6hdr);

		//tcphdr->dest = bpf_htons(80 + (ipv6hdr->saddr % nr_cpus));

	}
```

**Comment by lmb** [comment, marked as answer]
This question is not related to the library, so we won't be able to help you here. Please take a look at the [Guidelines](https://github.com/cilium/ebpf/discussions/886). You might have better luck on SO or Slack.
